llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskSchedulerService.java

```diff
@@ -429,6 +437,11 @@ public LlapTaskSchedulerService(TaskSchedulerContext taskSchedulerContext, Clock
     delayedTaskSchedulerExecutor =
         MoreExecutors.listeningDecorator(delayedTaskSchedulerExecutorRaw);
 
+    ExecutorService preemptTaskSchedulerExecutorRaw = Executors.newFixedThreadPool(1,
```

Comment:
If preemption is turned off, we won't need this executor.

Reply:
Well, I agree but the actual LLAP preemption Conf we are using https://github.com/apache/hive/blob/61d5c641b2e414c7b7dfd92f2b402db3583507c8/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java#L5023

is actually targeting the TaskExecutorService within the LlapDaemon (waitQueue tasks vs Running) and not the LlapTaskSchedulingService -- in a sense this a different type of preemption and I am not sure we should just use the same conf here.

Reply:
I checked that too and got confused. LlapTaskScheduler does the work of finding preemption candidates etc. even though preemption cannot occur in the end. Also, LlapTaskScheduler marks tasks as preempted and updates preemption stats eventhough nothing is preempted because of LLAP_DAEMON_TASK_SCHEDULER_ENABLE_PREEMPTION is false. Am I understanding this correctly?

This is not the problem of this patch obviously. I am just asking to understand. I'll +1 this regardless.